compiler/rustc_codegen_llvm/src/context.rs

```diff
@@ -290,6 +290,11 @@ pub(crate) unsafe fn create_module<'ll>(
         );
     }
 
+    // Disable ThinLTO if fat lto is requested. Otherwise lld defaults to thin lto.
```

Comment:
That sounds counterintuitive. Can you explain the relationship between the user's `lto` option and llvm's `lto` in the comments? 

And I think it needs a individual test to ensure that the previous `lto=far` option is not affected

Reply:
I changed the comment, is it clearer now?

(I **want to affect** the current `lto=fat` option, as it currently does thin lto, which I think is not intended and a bug :))

Reply:
I can confirm that the current behavior is surprising, since I ran into this issue with autodiff and it took multiple months to find out that despite selecting lto=fat there is still a thin-lto module in the compilation pipeline.